frontend/src/pages/registration/logInPage.js

```diff
@@ -1,43 +1,102 @@
-import React, { useState } from "react";
-import { Link } from "react-router-dom";
+import React, {useContext, useState} from "react";
+import { Link, useNavigate } from "react-router-dom";
+import axios from "axios";
+import apiEndpoints from "../../apiEndpoints";
+import { AuthContext } from "./AuthContext";
+import globalVariables from "../../globalVariables";
+import AuthBtn from "../../components/containers/authBtn";
+import {toast} from "sonner";
 
 
-function SignUpPage() {
-    function handleSubmit(e) {
-        e.preventDefault();
+function SignInPage() {
+    const authContext = useContext(AuthContext);
+    const navigate = useNavigate();
+    const [emailOrUserName, setEmailOrUserName] = useState('');
+    const [password, setPassword] = useState('');
+
+    const { login, isValidEmail, isValidUserName, isValidPassword } = authContext;
 
-        const form = e.target;
-        const formData = new FormData(form);
+    function isValidEmailOrUserName(emailOrUserName) {
+        if (!emailOrUserName.trim())
+            return false;
 
-        fetch('/some-api', { method: form.method, body: formData });
+        const isEmailRegex = /^[^@]$/
+        if (isEmailRegex.test(emailOrUserName))
+            return isValidEmail(emailOrUserName);
 
-        const formJson = Object.fromEntries(formData.entries());
-        console.log(formJson);
+        return isValidUserName(emailOrUserName);
     }
 
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
+    async function handleSubmit(e) {
+        e.preventDefault();
+
+        if (!isValidEmailOrUserName(emailOrUserName)) {
+            toast.error("Username or email can not be empty or contain such symbols!");
+            return;
+        }
+
+        if (!isValidPassword(password)) {
+            toast.error("Password must contain at least 8 symbols, where: 1 uppercase letter, 1 lowercase letter and 1 number!");
+            return;
+        }
+
+        try {
+            const response = await axios.post(
+                `${apiEndpoints.url}${apiEndpoints.login.login}`,
+                {
+                    email_or_username: emailOrUserName,
+                    password_hash: password,
+                    auth_provider: globalVariables.authStrategies.simpleStrategy
+                },
+                {
+                    headers: { 'Content-Type': 'application/json' },
+                }
+            );
+
+            login({ email: response?.data?.user?.email, username: response?.data?.user?.username });
+            toast.success(`You are successfully logged in!`);
```

Comment:
```suggestion
            toast.success(`You have successfully logged in`);
```

Though I am not sure whether we need to show that in the first place, but whatever